services/web-app/utils/use-id.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useEffect, useLayoutEffect, useState } from 'react'
+
+let serverHandoffComplete = false
+let globalIndex = 0
+// eslint-disable-next-line no-plusplus
+const genId = () => ++globalIndex
+
+// https://medium.com/@alexandereardon/uselayouteffect-and-ssr-192986cdcf7a
+const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayoutEffect : useEffect
+
+export const useId = (label) => {
+  /*
+   * If this instance isn't part of the initial render, we don't have to do the
+   * double render/patch-up dance. We can just generate the ID and return it.
+   */
+  const initialId = serverHandoffComplete ? genId() : null
+
+  const [id, setId] = useState(initialId)
+
+  useIsomorphicLayoutEffect(() => {
+    if (id === null) {
+      /*
+       * Patch the ID after render. We do this in `useLayoutEffect` to avoid any
+       * rendering flicker, though it'll make the first render slower (unlikely
+       * to matter, but you're welcome to measure your app and let us know if
+       * it's a problem).
+       */
+      setId(genId())
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (serverHandoffComplete === false) {
+      /*
+       * Flag all future uses of `useId` to skip the update dance. This is in
+       * `useEffect` because it goes after `useLayoutEffect`, ensuring we don't
+       * accidentally bail out of the patch-up dance prematurely.
+       */
+      serverHandoffComplete = true
+    }
+  }, [])
+  return `gtc__${label}__${id}`
+}
```

Comment:
Would love to better understand what's going on here. Otherwise looks good!

Reply:
it's using a prefix(gtc) + paramLabel + counter to generate unique ids: 
<img width="204" alt="image" src="https://user-images.githubusercontent.com/40550942/164271936-ca86248d-5adf-4938-9b87-64da7b911b55.png">
<img width="451" alt="image" src="https://user-images.githubusercontent.com/40550942/164271746-e7ff41da-05a9-4899-b7c6-15fa4cba94d3.png">

**changed prefix gtc(Gatsby Theme Platform) -> cp(Carbon Platform)